projects/js-packages/licensing/components/activation-screen/index.jsx

```diff
@@ -65,6 +65,30 @@ const ActivationScreen = props => {
 	const [ isSaving, setIsSaving ] = useState( false );
 	const [ activatedProduct, setActivatedProduct ] = useState( null );
 
+	const [ fetchingAvailableLicenses, setFetchingAvailableLicenses ] = useState( false );
+	const [ availableLicenses, setAvailableLicenses ] = useState( [] );
+
+	useEffect( () => {
+		const { apiRoot, apiNonce } = window?.myJetpackRest || {};
+		restApi.setApiRoot( apiRoot );
+		restApi.setApiNonce( apiNonce );
+
+		setFetchingAvailableLicenses( true );
+		return restApi
+			.getUserLicenses()
+			.then( ( { items } ) => {
```

Comment:
Instead of calling REST API directly here, I would suggest to make use of fetching the data via redux store, so that the data is re-usable elsewhere.

Here is what I could suggest to do:
- Rename `updateUserLicenses` to `fetchUserLicenses` in `projects/plugins/jetpack/_inc/client/state/licensing/actions.js`
- Dispatch the above `fetchUserLicenses()` action here inside `useEffect` or use a Query component
- Rename `getDetachedLicensesLoadingInfo` selector to `getIsLoadingLicenses` in `projects/plugins/jetpack/_inc/client/state/licensing/reducer.js`
- Use `getIsLoadingLicenses` selector in `projects/js-packages/licensing/components/activation-screen/index.jsx` to get the loading state.
- Use `getDetachedLicenses` selector to get the licenses.

Reply:
@manzoorwanijk, I initially thought of using redux to implement this new feature but I realized there is more things to setup on this package before I can do it in redux way. I actually noticed one package that has [reusable code](https://github.com/Automattic/jetpack/tree/trunk/projects/plugins/jetpack/_inc/client/state/licensing) I can probably reuse but I am not sure how to integrate it in the licensing package or is it even good idea to do it since it was structured  in a way it is only reusable within Jetpack plugin.

But now there is strong opinion to just implement it w/ redux I'll try and figure out how to reuse the code.